File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int record[26] = {0};
        for (char ch: magazine){
            record[ch - 'a']++;
        }
        for (char ch: ransomNote){
            record[ch - 'a']--;
        }
        for (int i = 0; i < 26; i++){
            if (record[i] < 0){
                return false;
            }
        }
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        hashmap = [0] * 26
        for c in magazine:
            hashmap[ord(c) - ord("a")] += 1
        
        for c in ransomNote:
            if hashmap[ord(c) - ord("a")] > 0:
                hashmap[ord(c) - ord("a")] -= 1
            else:
                return False
        return True
        